### PR TITLE
feat: Use Langfuse local to_openai_dict_format function to serialize messages

### PR DESCRIPTION
**Pull request by vblagoje**
## Why:
The Langfuse integration was using `ChatMessage.to_openai_dict_format()` from Haystack core, which has strict validation requirements (e.g., requiring ToolCall IDs). This led to several ChatGenerators that don't support tool call IDs failing during Langfuse tracing. To maintain stability and avoid potential breaking changes from upstream modifications to the core method, we need local control over ChatMessage serialization for Langfuse tracing.

- fixes https://github.com/deepset-ai/haystack-core-integrations/issues/1780

## What:
- **Core component**: Local `to_openai_dict_format` function in Langfuse tracer
- **Key features**: 
  - Less strict validation than core Haystack method (doesn't require ToolCall ID presence)
  - Maintains OpenAI API format compatibility for Langfuse ingestion
- **Integration**: Replaces calls to `ChatMessage.to_openai_dict_format()` with local implementation in span content tagging

## How can it be used:
- The change is internal to langfuse tracing - no user-facing API changes

## How did you test it:
- **Manual testing**: Verified with multiple chat generators to ensure proper serialization
- **Compatibility**: Confirmed OpenAI format compatibility for Langfuse ingestion
- **Edge cases**: Tested with tool calls, tool call results, and various message types using itinerary agent

## Notes for the reviewer:
- **Dependencies**: No new packages added - uses existing `json` module
- **Compatibility**: Maintains backward compatibility while providing more flexible serialization
- **Limitations**: Local implementation needs to be maintained separately from core Haystack changes, we now have duplication of code, albeit minimal
- **Risk mitigation**: Isolates Langfuse integration from potential breaking changes in core serialization logic

### CODE REVIEW

**Comment by vblagoje** [comment]
@mpangrazzi Seb and I talked about this one already internally. He has all the context already so I asked him to review

File: integrations/langfuse/src/haystack_integrations/tracing/langfuse/tracer.py [hunk 1]
```diff
@@ -64,6 +65,56 @@
 tracing_context_var: ContextVar[Dict[Any, Any]] = ContextVar("tracing_context")
 
 
+def to_openai_dict_format(chat_message: ChatMessage) -> Dict[str, Any]:
```
Comment (sjrl) [inline integrations/langfuse/src/haystack_integrations/tracing/langfuse/tracer.py]: @vblagoje what do you think of making this private for now. Since it's only internal to Langfuse and I think we'd prefer users to still use the `ChatMessage.to_openai_dict_format` in general.

File: integrations/langfuse/src/haystack_integrations/tracing/langfuse/tracer.py [hunk 2]
```diff
@@ -64,6 +65,56 @@
...
+    """
+    Convert a ChatMessage to the dictionary format expected by OpenAI's chat completion API.
```
Comment (sjrl) [inline integrations/langfuse/src/haystack_integrations/tracing/langfuse/tracer.py:70]: I'd update this docstring to mention the difference between this method and the `ChatMessage` util one.

**Review by sjrl — APPROVED** [review]
Looks good!
